lib/core/src/Cardano/Pool/Performance.hs

```diff
@@ -0,0 +1,278 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- A self contained module for ranking pools according to the delegation design
+-- spec.
+--
+-- * Definitions from Delegation Design Spec:
+--
+-- \tau -> fraction of the rewards that go to the treasury
+-- R -> total rewards from one epoch that will be split among stake-holders
+-- a_0 : [0, ∞) -> leader-stake influence on pool rewards
+-- z_0 = 1/k -> size of a saturated pool
+-- σ -> relative stake of pool (includes stake from leader)
+-- s -> relative stake of pool leader
+-- σ’  = min(σ, z_0)
+-- s’ = min(s, z_0)
+-- p_j_expected = n_j / max(N_expected, 1)
+-- \hat{f} -> pool rewards (includes average!)
+-- c -> pool costs in ADA
+-- m : [0, 1]  -> margin
+-- σ : [0,1]
```

Comment:
What about using these identifies consistently below instead of ad-hoc names like `saturedPoolRewards` ? Also, after a quick look, it looks like we truly need only a subset of these which we can categorize as follows:

```
Epoch Parameters
================

identifier   | domain / definition      | description
---          | ---                      | ---
R            | [0, ∞)                   | the total available rewards for the epoch (in ada).
a0           | [0, ∞)                   | owner-stake influence on pool rewards.
k            | [[0, ∞)                  | desired number of pools
z0           | 1/k                      | size of a saturated pool


Pool's Parameters
=================

identifier | domain / definition | description
---        | ---                 | ---
c          | [0, ∞)              | pool's costs
f          | [0, ∞)              | pool's rewards
m          | [0, 1]              | pool's margin
p_apparent | [0, 1]              | pool's apparent performance
s          | [0, ∞)              | relative stake of pool leader (i.e. pledge amount / total pool's stake)
σ          | [0, 1]              | relative stake of the pool, the fraction of the absolute stake that the pool controls
```

Reply:
Put it in actual haddock table syntax:

<img width="452" alt="Skärmavbild 2020-01-16 kl  18 07 43" src="https://user-images.githubusercontent.com/304423/72551758-ce42fa80-3895-11ea-884b-c686937f0f5a.png">

I realised one could argue this table to be superfluous as we have these definitions (mostly I believe) in the code itself, but might be nice to have 🤷‍♂